src/match.c

```diff
@@ -83,11 +83,15 @@ r_obj* ffi_locate_matches(r_obj* needles,
                           r_obj* nan_distinct,
                           r_obj* chr_proxy_collate,
                           r_obj* needles_arg,
-                          r_obj* haystack_arg) {
-  const struct vctrs_incomplete c_incomplete = parse_incomplete(incomplete);
-  const struct vctrs_no_match c_no_match = parse_no_match(no_match);
-  const struct vctrs_remaining c_remaining = parse_remaining(remaining);
-  const enum vctrs_multiple c_multiple = parse_multiple(multiple);
+                          r_obj* haystack_arg,
+                          r_obj* frame) {
+  struct r_lazy call = { .x = syms_call, .env = frame };
+  struct r_lazy internal_call = { .x = frame, .env = r_null };
+
+  const struct vctrs_incomplete c_incomplete = parse_incomplete(incomplete, internal_call);
+  const struct vctrs_no_match c_no_match = parse_no_match(no_match, internal_call);
+  const struct vctrs_remaining c_remaining = parse_remaining(remaining, internal_call);
+  const enum vctrs_multiple c_multiple = parse_multiple(multiple, internal_call);
```

Comment:
I've made a judgement call that these invalid argument issues should report `vec_locate_matches()` as the source of the error. I feel like _most_ of the time these arguments aren't directly exposed to the user. And even if they are, a suboptimal error here isn't the worst thing.